# U.S. Medical Insurance Costs

## Calculation #1 Average Insurance Cost 

In [1]:
#Average insurance charge 
#First I want to find out total amount of charges for the given sample
import csv
total_charges = []
with open("insurance.csv", "r") as insurance_csv:
    insurance_csv = csv.DictReader(insurance_csv)
    for row in insurance_csv:
        total_charges.append(row["charges"])
 
    


In [2]:
#After separating list of charges, I want to find the number of people given in the sample
num = len(total_charges)
print(num)

1338


In [3]:
#Sum of all charges
total_charges_int = [float(charge) for charge in total_charges]
total_charges_sum = 0
for element in range(0, len(total_charges_int)):
    total_charges_sum += total_charges_int[element]
print(total_charges_sum)
    

17755824.990759


In [4]:
#Now I can find the average insurance charge per person
average_charge = total_charges_sum / num
#rounding decimals
average_charge_r = round(average_charge, 2)
print(average_charge_r)

13270.42


## Calculation #2 Average Insurance Cost for Female and Male

In [5]:
#Separating insurance costs of females
female_charges_list = []
with open("insurance.csv", "r") as insurance_csv:
    insurance_csv = csv.DictReader(insurance_csv)
    for row in insurance_csv:
        if row["sex"] == "female":
            female_charges_list.append(row["charges"])
  
    
male_charges_list = []
with open("insurance.csv", "r") as insurance_csv:
    insurance_csv = csv.DictReader(insurance_csv)
    for row in insurance_csv:
        if row["sex"] == "male":
            male_charges_list.append(row["charges"])

    

In [6]:
#Next I am going to find total number of females and males in the sample
num_female = len(female_charges_list)
print(num_female)

num_male = len(male_charges_list)
print(num_male)

662
676


In [7]:
#Average insurance cost for females

female_charges_int = [float(charge) for charge in female_charges_list]
female_charges_sum = 0
for element in range(0, len(female_charges_int)):
    female_charges_sum += female_charges_int[element]
print(female_charges_sum)

average_female_charge = round(female_charges_sum / num_female, 2)
print(average_female_charge)


#Average insurance cost for males

male_charges_int = [float(charge) for charge in male_charges_list]
male_charges_sum = 0
for element in range(0, len(male_charges_int)):
    male_charges_sum += male_charges_int[element]
print(male_charges_sum)

average_male_charge = round(male_charges_sum / num_male, 2)
print(average_male_charge)
    
    
difference = abs(average_female_charge - average_male_charge)
print(difference)

8321061.194618994
12569.58
9434763.796139995
13956.75
1387.17


Calculation #2 conclusion: average insurance cost for men is higher than for women for 1387.17 USD in the given sample.

## Calculation #3 Average age of a person who has at least 3 children

In [8]:
age = []
with open("insurance.csv", "r") as insurance_csv:
    insurance_csv = csv.DictReader(insurance_csv)
    for row in insurance_csv:
        if int(row["children"]) >= 3:
            age.append(row["age"])
    
age_int = [int(age) for age in age]

total_age = 0
for i in range(0,len(age_int)):
    total_age += age_int[i]
    
num_people = len(age_int)
average_age = total_age /num_people
print(int(average_age))
    


40


Calculation #3 conclusion: Average age of people who have at least 3 children is 40 in the given sample.

## How many children do women who smoke have on average compared to non-smoker?

In [9]:
smoker_women = []
with open("insurance.csv", "r") as insurance_csv:
    insurance_csv = csv.DictReader(insurance_csv)
    for row in insurance_csv:
        if row["smoker"] == "yes" and row["sex"] == "female":
            smoker_women.append(row["children"])

children_int = [int(n) for n in smoker_women]

total_children = 0
for i in range(0,len(children_int)):
    total_children += children_int[i]

num_women = len(children_int)
print(num_women)

average_smoker_women_num = total_children / num_women
print(int(average_smoker_women_num))

#Smoker woman on average has 1 child.


nonsmoker_women = []
with open("insurance.csv", "r") as insurance_csv:
    insurance_csv = csv.DictReader(insurance_csv)
    for row in insurance_csv:
        if row["smoker"] == "no" and row["sex"] == "female":
            nonsmoker_women.append(row["children"])
            
children_nonsmoker = [int(n) for n in nonsmoker_women]

nonsmoker_num = len(children_nonsmoker)
print(nonsmoker_num)

nonsmoker_int = 0
for i in range(0,len(children_nonsmoker)):
               nonsmoker_int += children_nonsmoker[i]

average_children_nonsmoker = int(nonsmoker_int / nonsmoker_num)
print(average_children_nonsmoker)

#Nonsmoker woman on average has 1 child.

smoker_prop = (nonsmoker_num / (nonsmoker_num + num_women))
print(smoker_prop)
    

115
1
547
1
0.8262839879154078


There is no difference in number of children between smoker and nonsmoker women. However, there are considerable difference in number of smoker vs. nonsmokers. 83% of women in the given sample are nonsmokers.

## Insurance cost function

In [10]:
#I will extract each column from the csv file: age,sex,children

age_col = []
sex_col = []
bmi_col = []
children_col = []
smoker_col = []
region_col = []
charge_col = []

with open("insurance.csv", "r") as insurance_csv:
    insurance_csv = csv.DictReader(insurance_csv)
    for row in insurance_csv:
        age_col.append(row["age"])        

with open("insurance.csv", "r") as insurance_csv:
    insurance_csv = csv.DictReader(insurance_csv)
    for row in insurance_csv:
        sex_col.append(row["sex"])
    
with open("insurance.csv", "r") as insurance_csv:
    insurance_csv = csv.DictReader(insurance_csv)
    for row in insurance_csv:
        bmi_col.append(row["bmi"])
        
with open("insurance.csv", "r") as insurance_csv:
    insurance_csv = csv.DictReader(insurance_csv)
    for row in insurance_csv:
        children_col.append(row["children"])
        
with open("insurance.csv", "r") as insurance_csv:
    insurance_csv = csv.DictReader(insurance_csv)
    for row in insurance_csv:
        smoker_col.append(row["smoker"])
        
with open("insurance.csv", "r") as insurance_csv:
    insurance_csv = csv.DictReader(insurance_csv)
    for row in insurance_csv:
        charge_col.append(row["charges"])
        
insurance_details = list(zip(age_col,sex_col,bmi_col,children_col,smoker_col,charge_col))


#Next I am going to calculate estimated insurance cost using the insurance formula (250 * age - 128 * sex + 370 * bmi + 425 * num_of_children + 24000 * smoker - 12500)

def insurance_cost(insurance_details):
    for person in insurance_details:
        for i in range(len(person)):
            insurance_est = 250 * insurance_details[i][0] - 128 * insurance_details[i][1] + 370 * insurance_details[i][2] + 425 * insurance_details[i][3] + 24000 * insurance_details[i][4] - 12500
        difference = insurance_est - insurance_details[i][5]
        return difference

#print(insurance_cost(insurance_details))
    
def insurance_cost(age,sex,bmi,children,smoker,charge):
    insurance = 250 * age - 128 * sex + 370 * bmi + 425 * num_of_children + 24000 * smoker - 12500

        


#insurance_cost(age=age_col,sex=sex_col,bmi=bmi_col,children=children_col,smoker=smoker_col,charge=charge_col)
#insurance_cost(age_col,sex_col,bmi_col,children_col,smoker_col,charge_col)

